packages/application-tester/src/graphql-helper.ts

```diff
@@ -0,0 +1,146 @@
+import { ApolloClient } from 'apollo-client'
+import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory'
+import { HttpLink } from 'apollo-link-http'
+import fetch from 'cross-fetch'
+import { WebSocketLink } from 'apollo-link-ws'
+import { getMainDefinition } from 'apollo-utilities'
+import { ApolloLink, split } from 'apollo-link'
+import * as WebSocket from 'ws'
+import { SubscriptionClient } from 'subscriptions-transport-ws'
+import { ApolloClientOptions } from 'apollo-client/ApolloClient'
+import { ProviderTestHelper } from './provider-test-helper'
+
+type AuthToken = string | (() => string)
+
+export class GraphQLHelper {
+  constructor(private providerTestHelper: ProviderTestHelper) {}
+
+  public client(authToken?: AuthToken): ApolloClient<NormalizedCacheObject> {
+    return new ApolloClient({
+      cache: new InMemoryCache(),
+      link: this.getAuthLink(authToken).concat(this.getApolloHTTPLink()),
+      defaultOptions: {
+        query: {
+          fetchPolicy: 'no-cache',
+        },
+      },
+    })
+  }
+
+  /**
+   * IMPORTANT: After using this "DisconnectableApolloClient", you must call ".disconnect()" to close the socket. Otherwise
+   * it will keep waiting for messages forever
+   */
+  public async clientWithSubscriptions(authToken?: AuthToken): Promise<DisconnectableApolloClient> {
+    const subscriptionClient: SubscriptionClient = await this.subscriptionsClient(authToken)
+
+    const link = split(
+      ({ query }) => {
+        const definition = getMainDefinition(query)
+        return definition.kind === 'OperationDefinition' && definition.operation === 'subscription'
+      },
+      new WebSocketLink(subscriptionClient),
+      this.getAuthLink(authToken).concat(this.getApolloHTTPLink())
+    )
+
+    return new DisconnectableApolloClient(subscriptionClient, {
+      cache: new InMemoryCache(),
+      link: link,
+      defaultOptions: {
+        query: {
+          fetchPolicy: 'no-cache',
+        },
+      },
+    })
+  }
+
+  private getApolloHTTPLink(): HttpLink {
+    return new HttpLink({
+      uri: this.providerTestHelper.outputs.graphqlURL,
+      fetch,
+    })
+  }
+
+  private getAuthLink(authToken?: string | (() => string)): ApolloLink {
```

Comment:
I really don't understand what a type '() => string' is. Where could I find more information?

Reply:
Love you asked @Moneyba ! It is a type for a function that takes no parameters and returns a string. More information here: https://www.typescriptlang.org/docs/handbook/2/functions.html